src/tests/unit/tests/DetailsView/components/failure-instance-panel-control.test.tsx

```diff
@@ -209,25 +220,57 @@ describe('FailureInstancePanelControlTest', () => {
             .props()
             .primaryButtonOnClick(null);
 
+        clearPathSnippetDataMock.setup(handler => handler()).verifiable(Times.once());
         expect(wrapper.state().isPanelOpen).toBe(false);
 
         // This shouldn't be cleared because it stays briefly visible as the panel close animation happens
         expect(wrapper.state().currentInstance.failureDescription).toEqual(description);
 
         addInstanceMock.verifyAll();
+        clearPathSnippetDataMock.verifyAll();
+    });
+
+    test('componentDidUpdate reassigns state', () => {
+        const prevProps = createPropsWithType(CapturedInstanceActionType.CREATE);
+        const newProps = createPropsWithType(CapturedInstanceActionType.CREATE);
+        const newFailureInstance = {
+            failureDescription: null,
+            path: 'inputed path',
+            snippet: 'snippet for path',
+        };
+        newProps.failureInstance = newFailureInstance;
```

Comment:
For these cases where we modify a property after creation, we can sometimes use the spread operator (ellipses, e.g. …) to declare everything up front. For example, this might look like: 
```
const props = {
   failureInstance: YOUR_OBJECT_HERE,
   ...createPropsWithType(CapturedInstanceActionType.EDIT
};
``` 
It's nice when this works because the reader doesn't need to update their understanding of the object on a later line